keps/sig-node/127-user-namespaces/README.md

```diff
@@ -0,0 +1,393 @@
+ - secret
+ - downwardAPI
+ - emptyDir
+ - projected
+
+This list of volumes was chosen as they can't be used to share files with other
+pods.
+
+The mapping length will be 65535, mapping the range 0-65534 to the pod. This wide
+range makes sure most workloads will work fine.
+
+The mapping will be chosen by the kubelet, using a simple algorithm to give
+different pods in this category ("without" volumes) a non-overlapping mapping.
+Giving non-overlapping mappings generates the best isolation for pods.
+
+Furthermore, the node UID space of 2^32 can hold up to 2^16 pods each with a
+mapping length of 65535 (2^16-1) top. This imposes a limit of 65k pods per node,
+but that is not an issue we will hit in practice for a long time, if ever (today
+we run 110 pods per node by default).
+
+#### Phase 2: pods with volumes
+
+This phase makes user namespaces work in pods with volumes too. This is
+activated via the bool `pod.spec.HostUsers` too and pods fall into this mode if
+some other volume type than the ones listed for phase 1 is used. IOW, when phase
+2 is implemented, pods that use volume types not supported in phase 1, fall into
+the phase 2.
+
+All pods in this mode will use _the same_ mapping, chosen by the kubelet, with a
+length 65535, and mapping the range 0-65534 too. IOW, each pod will have its own user
+namespace, but they will map to _the same_ UIDs/GIDs in the host.
+
+Using the same mapping allows for pods to share files and mitigates all the
+listed vulnerabilities (as the host is protected from the container). It is also
+a trivial next-step to take, given that we have phase 1 implemented: just return
+the same mapping if the pod has other volumes.
+
+#### Phase 3: TBD
+
+### Summary of the Proposed Changes
+
+[This table](https://docs.google.com/presentation/d/1z4oiZ7v4DjWpZQI2kbFbI8Q6botFaA07KJYaKA-vZpg/edit#slide=id.gfd10976c8b_1_41) gives you a quick overview of each phase (note it is outdated, but still useful for a general overview).
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+TBD
+
+### Graduation Criteria
+
+##### Alpha
+- Phase 1 implemented
+
+##### Beta
+
+##### GA
+
+- Make plans on whether, when, and how to enable by default
+- Should we reconsider making the mappings smaller by default?
```

Comment:
at least this should move to beta criteria, I think.

Reply:
heh, moved to Beta now. Thanks!